Create basic obstacle avoidance sketch

FayonaC opened the pull request:
Created basic sketch to make the SmartCar stop moving forwards before it collides with an obstacle.

Code is taken from Dimitrios Platis' platform workshop 2020-04-01.

Resolves: #9

shonaighdouglas committed:
Create basic obstacle avoidance sketch

Created basic sketch to make the SmartCar stop moving forwards before it collides with an obstacle.

Code is taken from Dimitrios Platis' platform workshop 2020-04-01.

Resolves: #9

platisd commented on src/car/avoid/avoid.ino:
```diff
@@ -0,0 +1,25 @@
+
+void loop(){
+    unsigned int distance = sensor.getDistance();
+    if (distance && distance < 20){
```
It is considered a good practice unless a variable is a boolean, to be explicit about it. Namely in this case `distance != 0`

platisd reviewed (approved):
Minor (style) suggestion, otherwise it looks correct.
Just keep in mind that once there is no more obstacle in the way, the car will start moving again.

JenniNord reviewed (changes requested):
Make the suggested change before approval.

Commit:
Update basic obstacle avoidance sketch
Updated the if-statement in the loop method to specify that if the distance is not 0 and smaller than 20, the car will stop.

Resolves: #9

Victori4 reviewed (approved):
Good style improvement distance in if statement
